### PR TITLE
fix: fixing CNS IP releae for azure CNI in case of managed endpoint s…

### PR DESCRIPTION
**Pull request by behzad-mir**
**Issue Fixed**:
CNS is using [first 8 digit of containerID]-eth0 format to release IPs for azure CNI. However, when CNS Manage endpoint state is enable it should also use the full containerID as the key.

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [X] relevant PR labels added

**Notes**:

### CODE REVIEW

File: cns/restserver/ipam.go [hunk 1]
```diff
@@ -713,8 +713,13 @@ func (service *HTTPRestService) releaseIPConfigs(podInfo cns.PodInfo) error {
 	service.Lock()
 	defer service.Unlock()
 	ipsToBeReleased := make([]cns.IPConfigurationStatus, 0)
-
-	for i, ipID := range service.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+	key := podInfo.Key()
+	if _, isMapContainsKey := service.PodIPIDByPodInterfaceKey[podInfo.Key()]; !isMapContainsKey {
+		// the special case for azure CNI with managed endpoint state since the podInfo.Key() is in containeID-eth0 format and we need to use full ContainerID as well
```
Comment (tamilmani1989) [inline cns/restserver/ipam.go:718]: why can't we change podInfo.Key() format to containerid in same way as delegated azure ipam

Comment (nddq) [reply cns/restserver/ipam.go:718]: I was wondering this as well.. in Cilium azure-ipam we are setting `args.ContainerID` for both `PodInterfaceID` and `InfraContainerID`, is there a reason why we can't do it in stateless CNI?

Comment (behzad-mir) [reply cns/restserver/ipam.go]: @tamilmani1989 @nddq Addressed in the new commit

File: cns/service/main.go [hunk 2]
```diff
@@ -1244,6 +1244,9 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 		if err = PopulateCNSEndpointState(httpRestServiceImplementation.EndpointStateStore); err != nil {
 			return errors.Wrap(err, "failed to create CNS EndpointState From CNI")
 		}
+		if err = httpRestServiceImplementation.EndpointStateStore.Read(restserver.EndpointStoreKey, &httpRestServiceImplementation.EndpointState); err != nil {
+			return errors.Wrap(err, "Failed to restore endpoint state")
+		}
```
Comment (tamilmani1989) [inline cns/service/main.go:1249]: add comment on why this is required

**Review by tamilmani1989 — DISMISSED** [review]

File: cns/restserver/ipam.go [hunk 3]
```diff
@@ -427,6 +427,7 @@ func (service *HTTPRestService) removeEndpointState(podInfo cns.PodInfo) error {
 	service.Lock()
 	defer service.Unlock()
 	logger.Printf("[removeEndpointState] Removing endpoint state for infra container %s", podInfo.InfraContainerID())
+	logger.Printf("[removeEndpointState] Endpoint state dump: [%+v]", service.EndpointState)
```
Comment (rbtr) [inline cns/restserver/ipam.go:430]: let's not log the whole internal state in the hot path

Comment (behzad-mir) [reply cns/restserver/ipam.go]: addressed in the commit

**Review by rbtr — APPROVED** [review]
lgtm ty

**Review by nddq — APPROVED** [review]

File: cns/service/main.go [hunk 4]
```diff
@@ -1244,6 +1244,10 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 		if err = PopulateCNSEndpointState(httpRestServiceImplementation.EndpointStateStore); err != nil {
 			return errors.Wrap(err, "failed to create CNS EndpointState From CNI")
 		}
+		// endpoint state needs tobe loaded in memory so the subsequent Delete calls remove the state and release the IPs.
```
Comment (jpayne3506) [inline cns/service/main.go:1247]: Nit: spacing

**Review by jpayne3506 — APPROVED** [review]

**Pull request merged**